### PR TITLE
Dailymotion player component

### PR DESCRIPTION
**Pull request by zeno**
Add `amp-dailymotion` component to display a Dailymotion video. 

Include tests, documentation and example file.

### CODE REVIEW

File: examples/dailymotion.amp.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  <h4>Custom</h4>
+
+  <amp-dailymotion
+    data-videoid="x3rdtfy"
+    data-endscreen-enable="false"
+    data-sharing-enable="false"
+    data-ui-highlight="444444"
+    data-ui-logo="false"
+    data-ui-start_screen_info="false"
```
Comment (cramforce) [inline examples/dailymotion.amp.html]: The mixed `-` and `_` feels a bit weird.

File: extensions/amp-dailymotion/0.1/amp-dailymotion.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    /** @private {?Element} */
+    this.iframe_ = iframe;
+    return loadPromise(iframe);
+  }
+
+  /** @private */
+  addQueryParam_(param, query) {
+    const val = this.element.getAttribute(`data-${param}`);
+    if (val) {
+      query.push(`${param}=${encodeURIComponent(val)}`);
```
Comment (cramforce) [inline extensions/amp-dailymotion/0.1/amp-dailymotion.js]: Would you mind also encoding the param? Thanks!

**Comment by cramforce** [comment]
This looks great!

We've currently porting the video players to show a poster during pre-rendering (`amp-video` done, `amp-youtube` in progress). We should not do this in this change, but it would be great if we could do that in the future. It would require a predictable URL for a poster image of the video.

**Comment by zeno** [comment]
Thx for the feedback @cramforce. Just committed some changes.
Sure, showing poster during pre-rendering via a predictable URL to avoid extra request shouldn't be a problem.

**Comment by cramforce** [comment]
Looks great. Two more things

1. please add your file here https://github.com/ampproject/amphtml/blob/master/test/integration/test-example-validation.js#L55 and whitelist the error here https://github.com/ampproject/amphtml/blob/master/test/integration/test-example-validation.js#L72
2. Squash your commits to a single commit.

I added your component to #1732. It should start validating in the next few days.

**Comment by zeno** [comment]
Done.

Thanks @cramforce

**Comment by cramforce** [comment]
LGTM
Awesome! Will merge when travis is green.

**Comment by ndachez** [comment]
Hello , I tried to implement "amp- dailymotion" but I still encounter some errors. The video displays correctly... In Chrome console, I have 2 errors :

1. The attribute 'custom-element' in tag 'amp-access extension .js script' is set to the invalid value 'amp-dailymotion'
2. The tag 'amp-dailymotion' is disallowed.

The link : http://amp.chien.fr/actualite/cristina-cordula-m6-affirme-je-ne-suis-pas-contre-la-fourrure/#development=1

Thx.

**Comment by cramforce** [comment]
This should become valid within the day.

**Comment by ndachez** [comment]
Thx cramforce ! 

Indeed, I have the little message : "Powered by AMP ⚡ HTML – Version 1455904100660 / AMP validation successful."

Have a nice day :)

**Comment by okeul** [comment]
Hi !

This component works well but there are several errors in the console when the video is playing :
capture d ecran 2016-02-22 a 11 22 18 am

Example :
http://output.jsbin.com/gupojamowi/#development=1

**Pull request merged**